core/metacling/src/TClingMethodInfo.cxx

```diff
@@ -653,6 +678,7 @@ const char *TClingMethodInfo::TypeName() const
       // FIXME: Cint does not check!
       return nullptr;
    }
+   // The next *two* calls lock the interpreter mutex.
```

Comment:
Then we should take the lock here.  It is more efficient that lock/unlock twice.